--- ../src-base/minecraft/net/minecraftforge/fml/common/asm/transformers/AccessTransformer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/asm/transformers/AccessTransformer.java
@@ -45,7 +45,6 @@
 import net.minecraft.launchwrapper.IClassTransformer;
 import net.minecraftforge.fml.relauncher.FMLRelaunchLog;
 
-import org.apache.commons.io.IOUtils;
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassWriter;
 import org.objectweb.asm.tree.AbstractInsnNode;
@@ -470,8 +469,27 @@
         }
         finally
         {
-            IOUtils.closeQuietly(outJar);
-            IOUtils.closeQuietly(inJar);
+            if (outJar != null)
+            {
+                try
+                {
+                    outJar.close();
+                }
+                catch (IOException e)
+                {
+                }
+            }
+
+            if (inJar != null)
+            {
+                try
+                {
+                    inJar.close();
+                }
+                catch (IOException e)
+                {
+                }
+            }
         }
     }
     Multimap<String, Modifier> getModifiers()
