--- ../src-base/minecraft/net/minecraftforge/fml/common/asm/transformers/MarkerTransformer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/asm/transformers/MarkerTransformer.java
@@ -35,7 +35,6 @@
 
 import net.minecraft.launchwrapper.IClassTransformer;
 
-import org.apache.commons.io.IOUtils;
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassWriter;
 import org.objectweb.asm.tree.ClassNode;
@@ -270,8 +269,27 @@
         }
         finally
         {
-            IOUtils.closeQuietly(outJar);
-            IOUtils.closeQuietly(inJar);
+            if (outJar != null)
+            {
+                try
+                {
+                    outJar.close();
+                }
+                catch (IOException e)
+                {
+                }
+            }
+
+            if (inJar != null)
+            {
+                try
+                {
+                    inJar.close();
+                }
+                catch (IOException e)
+                {
+                }
+            }
         }
     }
 }
