--- ../src-base/minecraft/net/minecraftforge/fml/common/discovery/DirectoryDiscoverer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/discovery/DirectoryDiscoverer.java
@@ -33,7 +33,6 @@
 import net.minecraftforge.fml.common.ModContainerFactory;
 import net.minecraftforge.fml.common.discovery.asm.ASMModParser;
 
-import org.apache.commons.io.IOUtils;
 import org.apache.logging.log4j.Level;
 
 import com.google.common.base.Throwables;
@@ -74,14 +73,8 @@
             try
             {
                 FileInputStream fis = new FileInputStream(metadata);
-                try
-                {
-                    mc = MetadataCollection.from(fis, modDir.getName());
-                }
-                finally
-                {
-                    IOUtils.closeQuietly(fis);
-                }
+                mc = MetadataCollection.from(fis,modDir.getName());
+                fis.close();
                 FMLLog.fine("Found an mcmod.info file in directory %s", modDir.getName());
             }
             catch (Exception e)
@@ -108,11 +101,11 @@
             if (match.matches())
             {
                 ASMModParser modParser = null;
-                FileInputStream fis = null;
                 try
                 {
-                    fis = new FileInputStream(file);
+                    FileInputStream fis = new FileInputStream(file);
                     modParser = new ASMModParser(fis);
+                    fis.close();
                     candidate.addClassEntry(path+file.getName());
                 }
                 catch (LoaderException e)
@@ -122,12 +115,8 @@
                 }
                 catch (Exception e)
                 {
-                    throw Throwables.propagate(e);
+                    Throwables.propagate(e);
                 }
-                finally
-                {
-                    IOUtils.closeQuietly(fis);
-                }
 
                 modParser.validate();
                 modParser.sendToTable(table, candidate);
