--- ../src-base/minecraft/net/minecraftforge/client/model/ModelLoader.java
+++ ../src-work/minecraft/net/minecraftforge/client/model/ModelLoader.java
@@ -117,7 +117,6 @@
 
 public final class ModelLoader extends ModelBakery
 {
-    private static boolean firstLoad = Boolean.parseBoolean(System.getProperty("fml.skipFirstModelBake", "true"));
     private final Map<ModelResourceLocation, IModel> stateModels = Maps.newHashMap();
     private final Set<ModelResourceLocation> missingVariants = Sets.newHashSet();
     private final Map<ResourceLocation, Exception> loadingExceptions = Maps.newHashMap();
@@ -169,16 +168,6 @@
         HashMultimap<IModel, ModelResourceLocation> models = HashMultimap.create();
         Multimaps.invertFrom(Multimaps.forMap(stateModels), models);
 
-        if (firstLoad)
-        {
-            firstLoad = false;
-            for (ModelResourceLocation mrl : stateModels.keySet())
-            {
-                bakedRegistry.putObject(mrl, missingBaked);
-            }
-            return bakedRegistry;
-        }
-
         ProgressBar bakeBar = ProgressManager.push("ModelLoader: baking", models.keySet().size());
 
         for(IModel model : models.keySet())
