--- ../src-base/minecraft/net/minecraftforge/event/entity/player/PlayerEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/player/PlayerEvent.java
@@ -351,36 +351,4 @@
             return playerUUID;
         }
     }
-
-    /**
-     * Fired when the world checks if a player is near enough to be attacked by an entity.
-     * The resulting visibility modifier is multiplied by the one calculated by Minecraft (based on sneaking and more) and used to calculate the radius a player has to be in (targetDistance*modifier).
-     * This can also be used to increase the visibility of a player, if it was decreased by Minecraft or other mods. But the resulting value cannot be higher than the standard target distance.
-     */
-    public static class Visibility extends PlayerEvent
-    {
-
-        private double visibilityModifier = 1D;
-
-        public Visibility(EntityPlayer player)
-        {
-            super(player);
-        }
-
-        /**
-         * @param mod Is multiplied with the current modifier
-         */
-        public void modifyVisibility(double mod)
-        {
-            visibilityModifier *= mod;
-        }
-
-        /**
-         * @return The current modifier
-         */
-        public double getVisibilityModifier()
-        {
-            return visibilityModifier;
-        }
-    }
 }
