--- ../src-base/minecraft/net/minecraftforge/event/entity/player/PlayerDropsEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/entity/player/PlayerDropsEvent.java
@@ -21,9 +21,9 @@
 
 import java.util.List;
 
-import net.minecraftforge.common.ForgeHooks;
 import net.minecraftforge.fml.common.eventhandler.Cancelable;
 
+import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.entity.item.EntityItem;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.DamageSource;
@@ -48,7 +48,10 @@
      */
     public PlayerDropsEvent(EntityPlayer entity, DamageSource source, List<EntityItem> drops, boolean recentlyHit)
     {
-        super(entity, source, drops, ForgeHooks.getLootingLevel(entity, source.getEntity(), source), recentlyHit);
+        super(entity, source, drops,
+            (source.getEntity() instanceof EntityPlayer) ?
+                EnchantmentHelper.getLootingModifier(((EntityPlayer)source.getEntity())) : 0,
+            recentlyHit);
 
         this.entityPlayer = entity;
     }
