--- ../src-base/minecraft/net/minecraft/tileentity/TileEntityCommandBlock.java
+++ ../src-work/minecraft/net/minecraft/tileentity/TileEntityCommandBlock.java
@@ -23,8 +23,12 @@
     private boolean auto;
     private boolean conditionMet;
     private boolean sendToClient;
+    
     private final CommandBlockBaseLogic commandBlockLogic = new CommandBlockBaseLogic()
     {
+        {
+            sender = new org.bukkit.craftbukkit.command.CraftBlockCommandSender(this); // CraftBukkit - add sender
+        }
         public BlockPos getPosition()
         {
             return TileEntityCommandBlock.this.pos;
