--- ../src-base/minecraft/net/minecraftforge/fml/common/network/handshake/NetworkDispatcher.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/network/handshake/NetworkDispatcher.java
@@ -583,7 +583,7 @@
         if (!(cause instanceof ClosedChannelException))
         {
             // Mute the reset by peer exception - it's disconnection noise
-            if (cause.getMessage() != null && cause.getMessage().contains("Connection reset by peer"))
+            if (cause.getMessage().contains("Connection reset by peer"))
             {
                 FMLLog.log(Level.DEBUG, cause, "Muted NetworkDispatcher exception");
             }
