--- ../src-base/minecraft/net/minecraftforge/event/AttachCapabilitiesEvent.java
+++ ../src-work/minecraft/net/minecraftforge/event/AttachCapabilitiesEvent.java
@@ -126,8 +126,8 @@
      */
     public static class Item extends AttachCapabilitiesEvent<net.minecraft.item.Item>
     {
-        private final net.minecraft.item.ItemStack stack;
         @Deprecated
+        private final net.minecraft.item.ItemStack stack;
         private final net.minecraft.item.Item item;
         public Item(net.minecraft.item.Item item, net.minecraft.item.ItemStack stack)
         {
