--- ../src-base/minecraft/net/minecraftforge/fml/common/FMLModContainer.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/FMLModContainer.java
@@ -20,7 +20,6 @@
 
 import java.io.File;
 import java.io.FileInputStream;
-import java.io.InputStream;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
@@ -54,7 +53,6 @@
 import net.minecraftforge.fml.common.versioning.VersionRange;
 import net.minecraftforge.fml.relauncher.Side;
 
-import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.Level;
 
@@ -288,15 +286,7 @@
                 if (versionFile != null)
                 {
                     version = new Properties();
-                    InputStream sourceInputStream = source.getInputStream(versionFile);
-                    try
-                    {
-                        version.load(sourceInputStream);
-                    }
-                    finally
-                    {
-                        IOUtils.closeQuietly(sourceInputStream);
-                    }
+                    version.load(source.getInputStream(versionFile));
                 }
                 source.close();
             }
@@ -307,14 +297,8 @@
                 {
                     version = new Properties();
                     FileInputStream fis = new FileInputStream(propsFile);
-                    try
-                    {
-                        version.load(fis);
-                    }
-                    finally
-                    {
-                        IOUtils.closeQuietly(fis);
-                    }
+                    version.load(fis);
+                    fis.close();
                 }
             }
             return version;
