--- ../src-base/minecraft/net/minecraftforge/fml/common/ProgressManager.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/ProgressManager.java
@@ -32,6 +32,7 @@
 public class ProgressManager
 {
     private static final List<ProgressBar> bars = new CopyOnWriteArrayList<ProgressBar>();
+
     /**
      * Not a fully fleshed out API, may change in future MC versions.
      * However feel free to use and suggest additions.
@@ -56,9 +57,7 @@
         return bar;
     }
 
-    public static boolean isDisplayVSyncForced() {
-        return FMLCommonHandler.instance().isDisplayVSyncForced();
-    }
+
     /**
      * Not a fully fleshed out API, may change in future MC versions.
      * However feel free to use and suggest additions.
