--- ../src-base/minecraft/net/minecraftforge/fml/server/FMLServerHandler.java
+++ ../src-work/minecraft/net/minecraftforge/fml/server/FMLServerHandler.java
@@ -35,7 +35,6 @@
 import net.minecraft.util.IThreadListener;
 import net.minecraft.util.text.translation.LanguageMap;
 import net.minecraft.world.storage.SaveFormatOld;
-import net.minecraftforge.common.util.Java6Utils;
 import net.minecraftforge.fml.common.FMLCommonHandler;
 import net.minecraftforge.fml.common.FMLLog;
 import net.minecraftforge.fml.common.IFMLSidedHandler;
@@ -260,7 +259,15 @@
         finally
         {
             IOUtils.closeQuietly(stream);
-            Java6Utils.closeZipQuietly(zip);
+            try
+            {
+                if (zip != null)
+                    zip.close();
+            }
+            catch (IOException e)
+            {
+                // shush
+            }
         }
     }
 
@@ -332,10 +339,4 @@
     {
         // NOOP
     }
-
-    @Override
-    public boolean isDisplayVSyncForced()
-    {
-        return false;
-    }
 }
