--- ../src-base/minecraft/net/minecraftforge/fluids/DispenseFluidContainer.java
+++ ../src-work/minecraft/net/minecraftforge/fluids/DispenseFluidContainer.java
@@ -63,7 +63,7 @@
     private ItemStack fillContainer(IBlockSource source, ItemStack stack)
     {
         World world = source.getWorld();
-        EnumFacing dispenserFacing = source.getBlockState().getValue(BlockDispenser.FACING);
+        EnumFacing dispenserFacing = source.func_189992_e().getValue(BlockDispenser.FACING);
         BlockPos blockpos = source.getBlockPos().offset(dispenserFacing);
 
         ItemStack result = FluidUtil.tryPickUpFluid(stack, null, world, blockpos, dispenserFacing.getOpposite());
@@ -98,7 +98,7 @@
         }
 
         FluidStack fluidStack = fluidHandler.drain(Fluid.BUCKET_VOLUME, false);
-        EnumFacing dispenserFacing = source.getBlockState().getValue(BlockDispenser.FACING);
+        EnumFacing dispenserFacing = source.func_189992_e().getValue(BlockDispenser.FACING);
         BlockPos blockpos = source.getBlockPos().offset(dispenserFacing);
 
         if (fluidStack != null && fluidStack.amount == Fluid.BUCKET_VOLUME && FluidUtil.tryPlaceFluid(null, source.getWorld(), fluidStack, blockpos))
