--- ../src-base/minecraft/net/minecraftforge/fml/common/registry/VillagerRegistry.java
+++ ../src-work/minecraft/net/minecraftforge/fml/common/registry/VillagerRegistry.java
@@ -356,8 +356,8 @@
     {
         if (prof == null)
         {
-            if (entity.getZombieType() != ZombieType.NORMAL && entity.getZombieType() != ZombieType.HUSK)
-                entity.setZombieType(ZombieType.NORMAL);
+            if (entity.func_189777_di() != ZombieType.NORMAL && entity.func_189777_di() != ZombieType.HUSK)
+                entity.func_189778_a(ZombieType.NORMAL);
             return;
         }
 
@@ -369,13 +369,13 @@
 
         if (network >= 0 && network < 5) // Vanilla
         {
-            if (entity.getZombieType() == null || entity.getZombieType().getId() != network + 1)
+            if (entity.func_189777_di() == null || entity.func_189777_di().func_190150_a() != network + 1)
             {
-                entity.setZombieType(ZombieType.getVillagerByOrdinal(network));
+                entity.func_189778_a(ZombieType.func_190144_b(network));
             }
         }
-        else if (entity.getZombieType() != null)
-            entity.setZombieType(ZombieType.NORMAL);
+        else if (entity.func_189777_di() != null)
+            entity.func_189778_a(ZombieType.NORMAL);
     }
     public static void onSetProfession(EntityZombie entity, ZombieType type, int network)
     {
