--- ../src-base/minecraft/net/minecraft/item/crafting/RecipeBookCloning.java
+++ ../src-work/minecraft/net/minecraft/item/crafting/RecipeBookCloning.java
@@ -1,15 +1,21 @@
 package net.minecraft.item.crafting;
 
-import javax.annotation.Nullable;
 import net.minecraft.init.Items;
 import net.minecraft.inventory.InventoryCrafting;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.ItemWrittenBook;
 import net.minecraft.world.World;
+import org.bukkit.inventory.Recipe;
 
-public class RecipeBookCloning implements IRecipe
+import javax.annotation.Nullable;
+
+public class RecipeBookCloning /*extends ShapelessRecipes*/ implements IRecipe
 {
-    public boolean matches(InventoryCrafting inv, World worldIn)
+/*    public RecipeBookCloning() {
+		super(new ItemStack(Items.WRITTEN_BOOK, 0, -1), java.util.Arrays.asList(new ItemStack(Items.WRITABLE_BOOK, 0, 0)));
+	}*/
+
+	public boolean matches(InventoryCrafting inv, World worldIn)
     {
         int i = 0;
         ItemStack itemstack = null;
@@ -125,4 +131,9 @@
 
         return aitemstack;
     }
+
+    @Override
+    public Recipe toBukkitRecipe() {
+        return null;
+    }
 }
