--- ../src-base/minecraft/net/minecraftforge/client/ForgeHooksClient.java
+++ ../src-work/minecraft/net/minecraftforge/client/ForgeHooksClient.java
@@ -276,7 +276,7 @@
             for (int z = -distance; z <= distance; ++z)
             {
                 BlockPos pos = center.add(x, 0, z);
-                Biome biome = world.getBiome(pos);
+                Biome biome = world.getBiomeGenForCoords(pos);
                 int colour = biome.getSkyColorByTemp(biome.getFloatTemperature(pos));
                 r += (colour & 0xFF0000) >> 16;
                 g += (colour & 0x00FF00) >> 8;
