--- ../src-base/minecraft/net/minecraftforge/items/SlotItemHandler.java
+++ ../src-work/minecraft/net/minecraftforge/items/SlotItemHandler.java
@@ -43,24 +43,7 @@
     {
         if (stack == null)
             return false;
-
-        IItemHandler handler = this.getItemHandler();
-        ItemStack remainder;
-        if (handler instanceof IItemHandlerModifiable)
-        {
-            IItemHandlerModifiable handlerModifiable = (IItemHandlerModifiable) handler;
-            ItemStack currentStack = handlerModifiable.getStackInSlot(index);
-
-            handlerModifiable.setStackInSlot(index, null);
-
-            remainder = handlerModifiable.insertItem(index, stack, true);
-
-            handlerModifiable.setStackInSlot(index, currentStack);
-        }
-        else
-        {
-            remainder = handler.insertItem(index, stack, true);
-        }
+        ItemStack remainder = this.getItemHandler().insertItem(index, stack, true);
         return remainder == null || remainder.stackSize < stack.stackSize;
     }
 
@@ -88,30 +71,13 @@
     public int getItemStackLimit(ItemStack stack)
     {
         ItemStack maxAdd = stack.copy();
-        int maxInput = stack.getMaxStackSize();
-        maxAdd.stackSize = maxInput;
+        maxAdd.stackSize = maxAdd.getMaxStackSize();
+        ItemStack currentStack = this.getItemHandler().getStackInSlot(index);
+        ItemStack remainder = this.getItemHandler().insertItem(index, maxAdd, true);
 
-        IItemHandler handler = this.getItemHandler();
-        ItemStack currentStack = handler.getStackInSlot(index);
-        if (handler instanceof IItemHandlerModifiable) {
-            IItemHandlerModifiable handlerModifiable = (IItemHandlerModifiable) handler;
-
-            handlerModifiable.setStackInSlot(index, null);
-
-            ItemStack remainder = handlerModifiable.insertItem(index, maxAdd, true);
-
-            handlerModifiable.setStackInSlot(index, currentStack);
-
-            return maxInput - (remainder != null ? remainder.stackSize : 0);
-        }
-        else
-        {
-            ItemStack remainder = handler.insertItem(index, maxAdd, true);
-
-            int current = currentStack == null ? 0 : currentStack.stackSize;
-            int added = maxInput - (remainder != null ? remainder.stackSize : 0);
-            return current + added;
-        }
+        int current = currentStack == null ? 0 : currentStack.stackSize;
+        int added = maxAdd.stackSize - (remainder != null ? remainder.stackSize : 0);
+        return current + added;
     }
 
     @Override
