--- ../src-base/minecraft/net/minecraftforge/items/VanillaInventoryCodeHooks.java
+++ ../src-work/minecraft/net/minecraftforge/items/VanillaInventoryCodeHooks.java
@@ -50,7 +50,7 @@
                 for (int j = 0; j < dest.getSizeInventory(); j++)
                 {
                     ItemStack destStack = dest.getStackInSlot(j);
-                    if (dest.isItemValidForSlot(j, extractItem) && (destStack == null || destStack.stackSize < destStack.getMaxStackSize() && destStack.stackSize < dest.getInventoryStackLimit() && ItemHandlerHelper.canItemStacksStack(extractItem, destStack)))
+                    if (destStack == null || destStack.stackSize < destStack.getMaxStackSize() && ItemHandlerHelper.canItemStacksStack(extractItem, destStack))
                     {
                         extractItem = handler.extractItem(i, 1, false);
                         if (destStack == null)
